### PR TITLE
#157930785 Build out tests for the requests resource

### PR DESCRIPTION
**Pull request by denniswanjiru**
#### What does this PR do?
Developing the test cases for the  request resource

#### Description of Task to be completed?
- Create a new branch `ft-unittest-api-endpoints-157930785` from `challenge-2-develop` branch 
- Write test for the `/users/requests` endpoint [Method = GET]
- Write test for the `/users/requests` endpoint [Method = POST]
- Write test for the `/users/requests/<request_id>` endpoint [Method = GET]
- Write test for the `/users/requests/<request_id>` endpoint [Method = PUT]
- Write test for the `/users/requests/<request_id>` endpoint [Method = DELETE]

#### How should this be manually tested?
1. Clone the branch into your local machine as follows
2. Run the following commands on your terminal 

```
git clone -b ch-setup-api-env-157930602 git@github.com:DennisWanjiru/maintenance-tracker.git 

cd maintenance-tracker/app/

python3 -m virtualenv env -p python3.6

source env/bin/activate

pip install -r requirements.txt

nosetests
```

#### Any background context you want to provide?
Uses unittest and nose to run the tests

#### What are the relevant pivotal tracker stories?
[#157930785](https://www.pivotaltracker.com/story/show/157930785)

### CODE REVIEW

**Review by victoriaaoka — COMMENTED** [review]
- Nice tests though they are all happy.
-  Add tests for edge cases such as invalid user input.
- Test user authentication to ensure authorised users can create/view requests

**Comment by victoriaaoka** [comment]
LGTM

**Review by victoriaaoka — COMMENTED** [review]
However, you should minimize the number of assert statements you have per test. It is best practice to have your tests testing only one thing at a time.

File: test_requests_resource.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            "location": 234,
+        }, content_type="application/json")
+
+        self.assertEqual(empty_dict.status_code, 400)
+        self.assertEqual(empty_tuple.status_code, 400)
+        self.assertEqual(empty_list.status_code, 400)
+        self.assertEqual(empty_string.status_code, 400)
+        self.assertEqual(bad_data.status_code, 400)
+
+    def test_delete_a_request(self):
```
Comment (victoriaaoka) [inline test_requests_resource.py:174]: This test can be split into several tests: eg `test_delete_a_request` , `tests_delete_a_request_that_does_not_exist` etc

Comment (denniswanjiru) [reply test_requests_resource.py:174]: Thank you it makes a lot of sense because it gets cleaner and is readable and organized.  Working on it ASAP.
